src/expandableCalendar/WeekCalendar/index.js

```diff
@@ -48,23 +47,26 @@ class WeekCalendar extends Component {
 
     this.style = styleConstructor(props.theme);
 
+    this.presenter = new Presenter(props);
     this.list = React.createRef();
     this.page = NUMBER_OF_PAGES;
     // On Android+RTL there's an initial scroll that cause issues
     this.firstAndroidRTLScrollIgnored = !applyAndroidRtlFix;
 
     this.state = {
-      items: this.getDatesArray()
+      items: this.presenter.getDatesArray(this.props)
     };
   }
 
   componentDidUpdate(prevProps) {
-    const {updateSource, date} = this.props.context;
+    const {context} = this.props;
+    const {shouldComponentUpdate, getDatesArray, scrollToIndex, setProps} = this.presenter;
 
-    if (date !== prevProps.context.date && updateSource !== UPDATE_SOURCES.WEEK_SCROLL) {
-      this.setState({items: this.getDatesArray()});
-      this.list.current.scrollToIndex({animated: false, index: NUMBER_OF_PAGES});
+    if (shouldComponentUpdate(context, prevProps.context)) {
+      this.setState({items: getDatesArray(this.props)});
+      scrollToIndex(false);
     }
+    setProps(this.props);
```

Comment:
I still don't understand why you need the 'setProps()'.

Reply:
Fixed